crates/provider-sdk/src/provider.rs

```diff
@@ -632,6 +707,8 @@ pub async fn run_provider(
         commands,
         config,
         secrets: _secrets,
+        host_xkey,
```

Comment:
Same note here about maybe putting `public` in the name so it is clearer what the key is used for